akka-stream/src/main/scala/akka/stream/scaladsl/Framing.scala

```diff
@@ -224,7 +254,18 @@ object Framing {
     val lengthFieldLength:  Int,
     val lengthFieldOffset:  Int,
     val maximumFrameLength: Int,
-    val byteOrder:          ByteOrder) extends GraphStage[FlowShape[ByteString, ByteString]] {
+    val byteOrder:          ByteOrder,
+    computeFrameSize:       Option[(Array[Byte], Int) ⇒ Int]) extends GraphStage[FlowShape[ByteString, ByteString]] {
```

Comment:
Always having a function, and providing a cached (in the companion object) default `(_, n) => n` is better as it avoids the allocation of two lambdas when applying the function and then the `getOrElse` below. Alternatively, do not `map` but use a pattern match on the option instead (which desugars into `if-else`s).